src/PekatVisionSDK/instance.py

```diff
@@ -0,0 +1,427 @@
+            context_in_body: Whether to send the `context` `dict`  in the response body.
+                Set to `True` if the resulting context is larger than 4KB.
+            wait_for_init_model: Whether to wait for all models to be initialized before returning the [Instance][PekatVisionSDK.Instance] object.
+                Ignored if `already_running` is `True`.
+            ping: Whether to send a ping request to `HOST:PORT/ping` upon creating
+                the instance.
+            gpu: Which GPU to start project on.
+                Ignored if `already_running` is `True`.
+
+        Raises:
+            DistNotExistsError: If `dist_path` does not exist.
+            DistNotFoundError: If no PEKAT VISION was found in the default installation directory.
+            PortIsAllocatedError: If the specified port is already used by another process.
+            ValueError: If not `already_running` and project path is `None`.
+            RuntimeError: If the created process has no `stdout`.
+            ProjectNotFoundError: If the project was not found in the specified location.
+            PekatNotStartedError: If PEKAT VISION couldn't start a project.
+        """
+        self.project_path = Path(project_path).expanduser() if project_path else None
+        self.dist_path = Path(dist_path) if dist_path else None
+
+        self.host = host
+        self.already_running = already_running
+        self.disable_code = disable_code
+        self.tutorial_only = tutorial_only
+        self.context_in_body = context_in_body
+        self.wait_for_init_model = wait_for_init_model
+        self.gpu = gpu
+
+        self.stop_key: Optional[str] = None
+
+        if port is None:
+            self.port = self._find_free_ports()
+            self.port_is_defined = False
+        else:
+            self.port = port
+            self.port_is_defined = True
+
+        if not already_running:
+            self._start_instance()
+            atexit.register(self.stop)
+
+        if ping:
+            self.ping()
+
+        self.__stopping = False
+
+    def _get_dist_path(self) -> Path:
+        if self.dist_path:
+            if self.dist_path.exists():
+                return self.dist_path
+            raise DistNotExistsError(self.dist_path)
+
+        if platform.system() == "Windows":
+            pekat_install_parent_dir = Path(os.environ["PROGRAMFILES"])
+            pekat_install_base_name = "PEKAT VISION "
+        elif platform.system() == "Linux":
+            pekat_install_parent_dir = Path("/opt/PEKAT")
+            pekat_install_base_name = "pekat-vision-"
+        else:
+            raise DistNotFoundError
+
+        for p in sorted(
+            pekat_install_parent_dir.glob(f"{pekat_install_base_name}*"),
+            key=lambda x: version.parse(x.name.lstrip(pekat_install_base_name)),
+            reverse=True,
+        ):
+            return (
+                pekat_install_parent_dir / p
```

Comment:
Glob already keeps the parent directory name in the path, it shouldn't be needed to add it again.

Reply:
That makes sense, fixing.